### PR TITLE
Convert to extension handler

### PR DESCRIPTION
**Pull request by agoddard**

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+- [ ] Update README with any necessary configuration snippets
+
+- [ ] Binstubs are created if needed
+
+- [ ] RuboCop passes
+
+- [ ] Existing tests pass 
+
+#### New Plugins
```
Comment (majormoses) [inline .github/PULL_REQUEST_TEMPLATE.md:17]: We can kill this section no?

File: .travis.yml [hunk 2]
```diff
@@ -1,19 +1,33 @@
...
     recipients:
-      - sensu-plugin@sensu-plugins.io
+    - sensu-plugin@sensu-plugins.io
     on_success: change
     on_failure: always
-
 script:
-  - 'bundle exec rake default'
+- bundle exec rake default
+- gem build sensu-plugins-opsgenie.gemspec
```
Comment (majormoses) [inline .travis.yml:19]: extensions

File: .travis.yml [hunk 3]
```diff
@@ -1,19 +1,33 @@
...
+- gem install sensu-plugins-opsgenie-*.gem
```
Comment (majormoses) [inline .travis.yml:20]: extensions

File: .travis.yml [hunk 4]
```diff
@@ -1,19 +1,33 @@
...
+deploy:
+  provider: rubygems
+  api_key:
+    secure: K0vq7Ahny7bZIz1LnilosmNgHG/L4Nv6QuPaxo++/lQ7dWT9puRDLFw/pQI7+Xgb+LBLWFW2ye33q01qd1Cke8nuMKQduxnGeBzzH6fHJiXDcz5uEoKB2a8qdkwHHP3y9fisGaZgyhxlmui3fny3C0JjFizs9oNNYIt3MZcETTk=
```
Comment (majormoses) [inline .travis.yml:24]: will need someone with rubygems access to generate and populate.

File: .travis.yml [hunk 5]
```diff
@@ -1,19 +1,33 @@
...
+  gem: sensu-plugins-opsgenie
```
Comment (majormoses) [inline .travis.yml:25]: extensions

File: .travis.yml [hunk 6]
```diff
@@ -1,19 +1,33 @@
...
+  on:
+    tags: true
+    all_branches: true
+    rvm: 1.9.3
+    rvm: 2.0
+    rvm: 2.1
+    rvm: 2.2
+    repo: sensu-plugins/sensu-plugins-opsgenie
```
Comment (majormoses) [inline .travis.yml:33]: extensions

File: Rakefile [hunk 7]
```diff
@@ -1,24 +1,24 @@
...
 require 'yard/rake/yardoc_task'
 
 desc 'Don\'t run Rubocop for unsupported versions'
 begin
-  if RUBY_VERSION >= '2.0.0'
-    args = [:spec, :make_bin_executable, :yard, :rubocop]
-  else
-    args = [:spec, :make_bin_executable, :yard]
-  end
+  args = if RUBY_VERSION >= '2.0.0'
```
Comment (majormoses) [inline Rakefile:11]: we don't support ruby > 2 so we can clean this up.

File: Rakefile [hunk 8]
```diff
@@ -32,20 +32,16 @@ task :make_bin_executable do
...
-  version_file = 'lib/sensu-plugins-opsgenie/version.rb'
-
-  # Read the file, bump the PATCH version
-  contents = File.read(version_file).gsub(/(PATCH = )(\d+)/) { |_| Regexp.last_match[1] + (Regexp.last_match[2].to_i + 1).to_s }
-
-  # Write the new contents of the file
-  File.open(version_file, 'w') { |file| file.puts contents }
+desc 'Test for binstubs'
+task :check_binstubs do
+  bin_list = Gem::Specification.load('sensu-plugins-opsgenie.gemspec').executables
```
Comment (majormoses) [inline Rakefile:37]: extensions

**Review by majormoses — CHANGES_REQUESTED** [review]

**Comment by luisdavim** [comment]
Hi @agoddard can you please give some Feedback on @majormoses's comments?
Thanks.

**Comment by mbbroberg** [comment]
Hey @luisdavim, given this branch is on the repo, would you be okay with resolving this list of changes?

**Comment by majormoses** [comment]
As there is a lack of response/traction I'd say we should probably close this out or someone claim it. If no one claims this within the next week we should close the PR and create an issue for someone to claim in the future.

**Comment by majormoses** [comment]
closing due to lack of activity

**Pull request closed without merging**